src/ptc/util/jms.clj

```diff
@@ -75,6 +75,30 @@
          (map key->key)
          (into {}))))
 
+(defn update-cloud-path-keys
+  "If a file exists at the JMS cloud path, add the key to 'copy' otherwise
+  add the JMS key for the on-prem path to 'push'"
+  [wfl-key key-map workflow]
+  (let [[jms-cloud-key jms-on-prem-key] (get-in key-map [::push wfl-key])
+        on-prem-path (get workflow jms-on-prem-key)
+        cloud-path (get workflow jms-cloud-key)
+        uploaded? (try
+                 (misc/shell! "gsutil" "stat" cloud-path)
+                 (catch Exception _
+                   nil))]
+    (if (not uploaded?)
+      (misc/shell! "gsutil" "cp" on-prem-path cloud-path))
```

Comment:
Is this the behavior we want? Before we were uploading idats to a path constructed by `cloud-prefix`. Now we're uploading them to the path given to us by the JMS message. Personally I think it would be better to only upload to the `cloud-prefix` path, and only use the `cloud-path` if it was already copied up by zamboni.

Reply:
If we were given 50 messages all referencing the same `on-prem-path` and `cloud-path`, and the file doesn't start out already existing at the `cloud-path`, if we don't upload to the `cloud-path` itself would we hit overwrite issues on the subsequent 49 messages? Or would it just duplicate the file in different directories in our own bucket and we're cool with that? I think Iremember idats being pretty small.

Reply:
@pshapiro4broad, I thought we wanted to always give WFL the `cloud-path` and if it's not there PTC would upload it to that location?

Reply:
> If we were given 50 messages all referencing the same on-prem-path and cloud-path, and the file doesn't start out already existing at the cloud-path, if we don't upload to the cloud-path itself would we hit overwrite issues on the subsequent 49 messages?

That's a separate issue; and that could be true for any cloud path that push-to-cloud writes to, not just these paths.

Reply:
> @pshapiro4broad, I thought we wanted to always give WFL the cloud-path and if it's not there PTC would upload it to that location?

It's up to PTC/WFL to decide where a file goes in the cloud. We could keep using the GOTC paths but the I think the intent was to isolate the AOU arrays data into its own bucket, with its own set of access roles. Putting the data into the GOTC bucket doesn't seem to align with this goal.